format.go

```diff
@@ -350,3 +350,80 @@ func Glob(input, match string) bool {
 	// Check suffix last.
 	return trailingGlob || strings.HasSuffix(input, parts[last])
 }
+
+const maxWordSplitLength = 30
+
+// splitAtWord is a text splitter that takes into consideration a few things:
+//    * Ensuring the returned text is no longer than maxWidth.
+//    * Attempting to split at the closest word boundary, while still staying inside
+//      of the specific maxWidth.
+//    * if there is no good word boundry for longer words (or e.g. links, raw data, etc)
+//      that are above maxWordSplitLength characters, split the word into chunks to fit the
+// maximum width.
+func splitAtWord(input string, maxWidth int) (output []string) {
```

Comment:
This algorithm seems somewhat unnecessarily complicated to me, both in terms of computational complexity and code complexity. My proposal also supports word-boundary splitting but achieves this by checking if the given string exceeds the maximum length and if so looks for the last word separation character (i.e. `' '`) in this string starting at the end of the string and splits it there. If such a word separation character is not found it simply splits at the next valid UTF-8 rune.

https://github.com/lrstanley/girc/blob/0503d81bfbf5740eedd6b3c4cb2b94e3ee6382b1/split.go#L36-L49

Just for the learning experience I would be interested to know what's wrong with my splitting algorithm. You probably have your reasons for employing your own :wink:

Reply:
The word-splitting code I have isn't the best (and I'd still like to clean it up), but I think the approach makes the most sense. The idea isn't to split just on space and if you can't do that, split at the last character.

You do not want to split in the middle of a word if you can, the idea is to ONLY split if the word (or other object inside of the string) is so long it only makes sense to split it. Otherwise, you're splitting the simplest of things ("and" becomes "a" & "nd").

The approach I take is similar to how most word editors, code editors, browsers, etc, split paragraphs. Even the text box that I am currently typing in, in GitHub does.

Reply:
I may be misunderstanding your implementation, but, regardless, just keep in mind this isn't the end result. Things like colors/backgrounds/bold/italic, as well as special treatment for things like URL's still need to be taken into count.